src/main/java/de/tum/in/www1/artemis/service/exam/StudentExamService.java

```diff
@@ -236,25 +236,36 @@ else if (exercise instanceof ModelingExercise) {
     }
 
     /**
-     * Assess the modeling-, text and file upload exercises of student exams of an exam which are not submitted with 0 points.
+     * Assess all exercises, except quiz exercises, of student exams of an exam which are not submitted with 0 points.
      *
      * @param exam the exam
      * @param assessor the assessor should be the instructor making the call.
      * @return returns the set of unsubmitted StudentExams, the participations of which were assessed
      */
     public Set<StudentExam> assessUnsubmittedStudentExams(final Exam exam, final User assessor) {
-        // TODO Simon Entholzer: we should also do this for programming exercises with manual assessment
         Set<StudentExam> unsubmittedStudentExams = studentExamRepository.findAllUnsubmittedWithExercisesByExamId(exam.getId());
         Map<User, List<Exercise>> exercisesOfUser = unsubmittedStudentExams.stream()
-                .collect(Collectors.toMap(StudentExam::getUser,
-                        studentExam -> studentExam.getExercises().stream()
-                                .filter(exercise -> exercise instanceof ModelingExercise || exercise instanceof TextExercise || exercise instanceof FileUploadExercise)
-                                .collect(Collectors.toList())));
+                .collect(
+                        Collectors
+                                .toMap(StudentExam::getUser,
+                                        studentExam -> studentExam.getExercises().stream().filter(exercise -> exercise instanceof ModelingExercise
+                                                || exercise instanceof TextExercise || exercise instanceof FileUploadExercise || exercise instanceof ProgrammingExercise)
+                                                .collect(Collectors.toList())));
+
```

Comment:
Can you refactor this into a small method? This code is duplicated in line 295.

Reply:
I put all this code in one method which does it